ui/components/app/user-preferenced-currency-display/user-preferenced-currency-display.component.js

```diff
@@ -86,3 +90,6 @@ UserPreferencedCurrencyDisplay.propTypes = {
    */
   ...CurrencyDisplay.propTypes,
```

Comment:
This seems to be breaking the type inference... I have noticed that for some reason using `Object.assign({}, UserPreferencedCurrencyDisplayPropTypes, CurrencyDisplay)` works... but ends up breaking linting for functional component opts destruct.
Probably the best way to deal with this is to be explicit about the propTypes, instead of extending from a different component propTypes. For example `UserPreferencedCurrencyDisplayPropTypes` would look like this:
```js
const UserPreferencedCurrencyDisplayPropTypes = {
  className: PropTypes.string,
  'data-testid': PropTypes.string,
  prefix: PropTypes.string.isRequired,
  value: PropTypes.string,
  numberOfDecimals: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  hideLabel: PropTypes.bool,
  hideTitle: PropTypes.bool,
  style: PropTypes.object,
  showEthLogo: PropTypes.bool,
  ethLogoHeight: PropTypes.number,
  type: PropTypes.oneOf([PRIMARY, SECONDARY]),
  ethNumberOfDecimals: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  fiatNumberOfDecimals: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  showFiat: PropTypes.bool,
  showNative: PropTypes.bool,
  showCurrencySuffix: PropTypes.bool,
  currency: PropTypes.string,
  denomination: PropTypes.oneOf([
    EtherDenomination.GWEI,
    EtherDenomination.ETH,
  ]),
  displayValue: PropTypes.string,
  prefixComponent: PropTypes.node,
  suffix: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  prefixComponentWrapperProps: PropTypes.object,
  textProps: PropTypes.object,
  suffixProps: PropTypes.object,
};
```

Reply:
This pattern was introduced by the another PR. I propose we fix this in another PR (unless @jpuri is fine with fixing it here).

Reply:
@segun although it was introduced in a different PR, it is this PR that breaks type inference. And because of that, it should be fixed in this PR.

It breaks type inference, because on this PR we have introduced `PropTypes.InferProps<typeof propTypes>>` as way for TS to properly infer the functional component props type from propTypes (see [here](https://github.com/MetaMask/metamask-extension/pull/23260/files#diff-c99d34b0461ac441cee8391fdf35e9622ab2a18c19820f748a2536a89f91d2c7R13)). Allowing us to benefit from TS without having to convert all jsx into tsx.

Reply:
Good point I made the update.